File: mystery/nombre_mystere.py

import random


def nombre_mystere():
    nombre_aleatoire = random.randint(1, 100)
    nombre_tentatives = 5
    na_pas_trouve = True
    while na_pas_trouve:
        entree = int(input('Choisir un nombre entre 1 et 100 : '))

        if (nombre_aleatoire > entree):
            nombre_tentatives = nombre_tentatives - 1
            if nombre_tentatives == 0:
                na_pas_trouve = False
            print("C'est plus il vous reste : " +
                  str(nombre_tentatives) + ' tentatives')

        elif (nombre_aleatoire < entree):
            nombre_tentatives = nombre_tentatives - 1
            if nombre_tentatives == 0:
                na_pas_trouve = False
            print("C'est moins il vous reste : " +
                  str(nombre_tentatives) + ' tentatives')

        else:
            print('Mes félicitations vous avez trouvé la bonne reponse : ', entree)
            na_pas_trouve = False

File: mystery/nombreMystere.js
function generateNumber() {
    return Math.floor(Math.random() * 100);
  }
  
  function nombreMystere() {
    const generee = generateNumber();
    let nombreTentative = 5;
    let naPasTrouve = true;
  
    do {
      let entree = parseInt(prompt('Entrez un nombre compris entre 0 et 100'));
  
      if (entree === generee) {
        alert(`Mes félicitations vous avez trouvé la bonne reponse : ${generee}`);
        naPasTrouve = false;
      } else {
        nombreTentative -= 1;
        if (entree > generee) {
          alert("C'est moins ");
        } else {
          alert("C'est plus");
        }
      }
  
      if (nombreTentative === 0) {
        alert('Vous avez perdu ');
        naPasTrouve = false;
      }
    } while (naPasTrouve);
  
    //   while (entree !== generee) {
    //     if (nombreTentative === 0) {
    //       alert('Vous avez perdu ');
    //       break;
    //     }
    //     if (entree > generee) {
    //       alert("C'est moins ");
    //     } else if (entree < generee) {
    //       alert("C'est plus");
    //     } else {
    //       alert(`Mes félicitations vous avez trouvé la bonne reponse : ${generee}`);
    //       break;
    //     }
    //     nombreTentative -= 1;
    //     entree = parseInt(prompt('Entrez un nombre compris entre 0 et 100'));
    //   }
  }
  nombreMystere();
  